### DISCUSSION TITLE
rancher,gitlab e renovatebot

### DISCUSSION THREAD

**Discussion by Rbsn-joses** [renovatebot/renovate#7954] [Request Help]
Hello, I would like to know how i could  implement the renovate bot on rancher.
what are its environment variables, how would i set up?

**Comment by rarkins** [comment]
Can you give more details about what you mean, or perhaps an example?

Does Rancher use its own proprietary file format to specify dependencies, or does it reuse others like k8s manifests?

**Comment by Rbsn-joses** [comment]
It would be more like an API container in the rancher that would send a merge request, so it would need some environment variables to be configured and thus communicate with gitlab self-hosted

**Comment by viceice** [comment]
@Rbsn-joses Do you like to run renovate with Rancher v1.x or v2.x?

**v1.x**
Rancher v1 is deprecated and uses docker compose based config files to run renovate. There is no cron job, so better use gitlab pipeline to run renovate, check [this](https://gitlab.com/renovate-bot/renovate-runner/-/blob/master/README.md) out for a simple sample.

**v2.x**
Rancher v2 uses kubernets, so use [this](https://docs.renovatebot.com/self-hosting/#kubernetes) or better use gitlab pipeline, see above.
